DiskIOMeter.c

```diff
@@ -36,20 +36,17 @@ static void DiskIOMeter_updateValues(Meter* this) {
    static uint64_t cached_last_update;
    uint64_t passedTimeInMs = pl->realtimeMs - cached_last_update;
 
-   /* update only every 500ms */
+   /* update only every 500ms to have a sane span for rate calculation */
    if (passedTimeInMs > 500) {
       static uint64_t cached_read_total;
       static uint64_t cached_write_total;
       static uint64_t cached_msTimeSpend_total;
       uint64_t diff;
 
-      cached_last_update = pl->realtimeMs;
-
       DiskIOData data;
-
-      hasData = Platform_getDiskIO(&data);
-      if (!hasData) {
-         this->values[0] = 0;
+      status = Platform_getDiskIO(&data) ? (cached_last_update == 0 ? S_INIT : S_DATA) : S_NODATA;
```

Comment:
This is not the easiest LOC to read. What about writing the if blocks explicitly?